checkups/kubevirt-vm-latency/vmlatency/internal/checkup/checkup_test.go

```diff
@@ -156,6 +157,84 @@ func TestCheckupSetupShouldCreateVMsWith(t *testing.T) {
 	}
 }
 
+func TestCheckupSetupShouldCreateVMsWithPodAntiAffinity(t *testing.T) {
+	t.Run("when source and target nodes names are not specified", func(t *testing.T) {
+		testClient := newTestClient()
+		testClient.returnNetAttachDef = newTestNetAttachDef("")
+		testCheckup := checkup.New(testClient, testNamespace, config.CheckupParameters{}, &checkerStub{})
+
+		assert.NoError(t, testCheckup.Preflight())
+		assert.NoError(t, testCheckup.Setup(context.Background()))
+		assertSourceVmiPodAntiAffinityExist(t, testClient)
+		assertTargetVmiPodAntiAffinityExist(t, testClient)
+		assertVmiNodeAffinityNotExist(t, testClient, checkup.SourceVmiName)
+		assertVmiNodeAffinityNotExist(t, testClient, checkup.TargetVmiName)
+	})
+}
+
+func TestCheckupSetupShouldCreateVMsWithNodeAffinity(t *testing.T) {
+	t.Run("when both source and target node names are specified", func(t *testing.T) {
+		const (
+			testSourceNode = "source.example.com"
+			testTargetNode = "target.example.com"
+		)
+		testClient := newTestClient()
+		testClient.returnNetAttachDef = newTestNetAttachDef("blah")
+		testCheckupParams := config.CheckupParameters{SourceNodeName: testSourceNode, TargetNodeName: testTargetNode}
+		testCheckup := checkup.New(testClient, testNamespace, testCheckupParams, &checkerStub{})
+
+		assert.NoError(t, testCheckup.Preflight())
+		assert.NoError(t, testCheckup.Setup(context.Background()))
+		assertVmiNodeAffinityExist(t, testClient, checkup.SourceVmiName, testSourceNode)
+		assertVmiNodeAffinityExist(t, testClient, checkup.TargetVmiName, testTargetNode)
+		assertVmiPodAntiAffinityNotExist(t, testClient, checkup.SourceVmiName)
+		assertVmiPodAntiAffinityNotExist(t, testClient, checkup.TargetVmiName)
+	})
+}
+
+func assertSourceVmiPodAntiAffinityExist(t *testing.T, testClient *clientStub) {
+	actualVmi, err := testClient.GetVirtualMachineInstance(testNamespace, checkup.SourceVmiName)
+	assert.NoError(t, err)
+
+	assert.NotNil(t, actualVmi.Spec.Affinity.PodAntiAffinity)
+	expectedPodAntiAffinity := vmi.NewPodAntiAffinity(vmi.Label{Key: checkup.LabelLatencyCheckVM, Value: ""})
+	assert.Equal(t, expectedPodAntiAffinity, actualVmi.Spec.Affinity.PodAntiAffinity)
+}
+
+func assertTargetVmiPodAntiAffinityExist(t *testing.T, testClient *clientStub) {
+	actualVmi, err := testClient.GetVirtualMachineInstance(testNamespace, checkup.TargetVmiName)
+	assert.NoError(t, err)
+
+	assert.NotNil(t, actualVmi.Spec.Affinity.PodAntiAffinity)
+	expectedPodAntiAffinity := vmi.NewPodAntiAffinity(vmi.Label{Key: checkup.LabelLatencyCheckVM, Value: ""})
+	assert.Equal(t, expectedPodAntiAffinity, actualVmi.Spec.Affinity.PodAntiAffinity)
+}
```

Comment:
Is it necessary to have this separation now when we are using a single label?
```suggestion
func assertVmiPodAntiAffinityExist(t *testing.T, testClient *clientStub, vmiName string) {
	actualVmi, err := testClient.GetVirtualMachineInstance(testNamespace, vmiName)
	assert.NoError(t, err)

	assert.NotNil(t, actualVmi.Spec.Affinity.PodAntiAffinity)
	expectedPodAntiAffinity := vmi.NewPodAntiAffinity(vmi.Label{Key: checkup.LabelLatencyCheckVM, Value: ""})
	assert.Equal(t, expectedPodAntiAffinity, actualVmi.Spec.Affinity.PodAntiAffinity)
}
```

Reply:
Done